crates/header-translator/src/cache.rs

```diff
@@ -119,6 +119,36 @@ impl<'a> Cache<'a> {
     }
 
     fn update_file(&self, file: &mut File) {
+        // disambiguate duplicate names
+        let mut names = BTreeMap::<(ItemIdentifier, String), &mut Method>::new();
+        for stmt in file.stmts.iter_mut() {
+            match stmt {
+                Stmt::Methods { cls, methods, .. } => {
+                    for this in methods.iter_mut() {
+                        let key = (cls.clone(), this.fn_name.clone());
+                        if let Some(that) = names.get_mut(&key) {
+                            that.fn_name = that.selector.replace(':', "_");
+                            this.fn_name = this.selector.replace(':', "_");
+                        } else {
+                            names.insert(key, this);
+                        }
+                    }
+                }
+                Stmt::ProtocolDecl { id, methods, .. } => {
+                    for this in methods.iter_mut() {
+                        let key = (id.clone(), this.fn_name.clone());
+                        if let Some(that) = names.get_mut(&key) {
+                            that.fn_name = that.selector.replace(':', "_");
+                            this.fn_name = this.selector.replace(':', "_");
+                        } else {
+                            names.insert(key, this);
+                        }
+                    }
+                }
```

Comment:
You can avoid the duplication by matching like this:
```rust
Stmt::Methods { cls: id, methods, .. }
| Stmt::ProtocolDecl { id, methods, .. } =>
```

Reply:
Ah, good point!